[Bounty] Fixes portal accessory behaviour

Azarak opened the pull request:
<!-- FILLING OUT THIS FORM PROPERLY AND ADDING A PROPER CHANGELOG SECTION IS **NECESSARY** FOR PULL REQUESTS. IF THE INFORMATION IS NOT PROVIDED YOUR PR WILL NOT BE CONSIDERED FOR MERGING -->

<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

# About The Pull Request
Implemented to do the bounty https://github.com/SPLURT-Station/S.P.L.U.R.T-Station-13/issues/779
This fixes portal fleshlight sex behaviour to cum inside your partner.

The fix was implemented by adding a new argument to the sex proc chain, and using that as a basis for a determining whether an entity will cum inside, rather than FALSE. It still becomes overriden by the existing logic if the logic desires - no other behaviours are changed.

This argument can be also used to tweak other interactions easily

![dreamseeker_gr5e4cbGPf](https://user-images.githubusercontent.com/6381979/223803186-fea574cc-9b6d-4fa4-bd72-7aa7730f889d.png)
![dreamseeker_w2kHZMIBJM](https://user-images.githubusercontent.com/6381979/223803204-2e18ff20-e901-4de0-8e04-e1b083852c29.png)

Update:
I have fixed weird arguments with the `mob_climax_partner` proc, which could be a cause for a runtime.
I have added an anonymity feature to the climax proc chain, and made the portal fleshlights use it
The anonymity feature obstructs information about who climaxed into who in messages, to faciliate portal fleshlight intended behaviour

![dreamseeker_aK7FuumwlY](https://user-images.githubusercontent.com/6381979/226105042-d8cabb20-63fa-4375-a5ce-5ae5947a0e77.png)




<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->
<!-- If your PR is related to one of our discord suggestions, please add the number of this suggestion to this section. Or if you may, the message link of said suggestion-->

## Why It's Good For The Game
Bugs are bad

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## A Port?
No

<!-- Just say if it is a port of something and link the original pr/commit/whatever. -->

## Pre-Merge Checklist
<!-- Don't bother filling these in while creating your Pull Request, just click the checkboxes after the Pull Request is opened and you are redirected to the page. -->
- [x] You tested this on a local server.
- [x] This code did not runtime during testing.
- [x] You documented all of your changes.
<!-- Neither the compiler nor workflow checks are perfect at detecting runtimes and errors. It is important to test your code/feature/fix locally. -->

## Changelog

:cl:
fix: Portal fleshlight orgasms will now cum inside your target
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

BongaTheProto commented:
could you maybe mark the edits made to `modular_sand/code/datums/interactions/lewd_definitions.dm` as SPLURT EDIT so we more easily know that those lines have been edited?

Azarak commented:
Someone pointed to me that this failed to work - I have investigated and found out an error with the arguments for `mob_climax_partner` which could be responsible for a runtime

I have fixed the arguments in the procs/calls to that

I have also been notified that there should be anonymity regarding who is using the device when they climax, I have added such feature aswell

BongaTheProto commented on modular_sand/code/datums/interactions/lewd_definitions.dm:
```diff
@@ -761,15 +772,27 @@
 				LAZYADD(obscure_to, src)
 	else //todo: better self cum messages
 		message = "cums all over themselves!"
+
+	if(partner_carbon_check && cumin)
+		switch(target_orifice)
+			if(CUM_TARGET_VAGINA)
+				target_gen = c_partner.getorganslot(ORGAN_SLOT_VAGINA)
+			if(CUM_TARGET_ANUS)
+				target_gen = c_partner.getorganslot(ORGAN_SLOT_ANUS)
+			if(CUM_TARGET_BREASTS)
+				target_gen = c_partner.getorganslot(ORGAN_SLOT_BREASTS)
+			if(CUM_TARGET_PENIS)
+				target_gen = c_partner.getorganslot(ORGAN_SLOT_PENIS)
+
```
I believe this is already handled on the interactions themselves

BongaTheProto replied:
> Only in 2 cases, and only for the female genital. And those 2 cases should be removed in favor of this because this is a better way of doing this, unless you disagree

I suppose that's fair, though remember it's a SPLURT addition so you might wanna check it also doesn't cause issues with the code already in place, also the override in modular_splurt

BongaTheProto commented:
Please fix the merge conflicts

BongaTheProto commented:
So, the problem with this pr is that it changes a lot of nonmodular code in order to handle the portal underwear behaviour, which are quite intrusive and snowflakey at that. In which case it'd be a better option to either keep the changed to modular/portal panties files and code or port it to upstream

Also, it still has the changes we discussed in discord as not good for this pr, which should be removed
